File: yggdrasil/examples/formatted_io4/src/formatted_io4_modelA.R
# Import library for input/output
library(yggdrasil)

# Initialize input/output channels
in_channel <- YggInterface('YggPandasInput', 'inputA')
out_channel <- YggInterface('YggPandasOutput', 'outputA')

# Loop until there is no longer input or the queues are closed
while(TRUE) {

  # Receive input from input channel
  # If there is an error, the flag will be False
  c(flag, frame) %<-% in_channel$recv()
  if (!flag) {
    print('Model A: No more input.')
    break
  }

  # Print received message
  fprintf('Model A: (%d rows)', nrow(frame))
  print(frame)

  # Send output to output channel
  # If there is an error, the flag will be False
  flag = out_channel$send(frame)
  if (!flag) {
    stop('Model A: Error sending output.')
  }

}

File: yggdrasil/examples/formatted_io4/src/formatted_io4_modelA.jl
# Import library for input/output channels
using Yggdrasil
using Printf

# Initialize input/output channels
in_channel = Yggdrasil.YggInterface("YggPandasInput", "inputA")
out_channel = Yggdrasil.YggInterface("YggPandasOutput", "outputA")

# Loop until there is no longer input or the queues are closed
while true

    # Receive input from input channel
    # If there is an error, the flag will be false
    flag, frame = in_channel.recv()
    if (!flag)
        println("Model A: No more input.")
	break
    end

    # Print received message
    # nrows = len(frame.index)
    # @printf("Model A: (%d rows)\n", nrows)
    println(frame)

    # Send output to output channel
    # If there is an error, the flag will be false
    flag = out_channel.send(frame)
    if (!flag)
        error("Model A: Error sending output.")
    end

end  # while